libultraship/libultraship/Lib/Fast3D/gfx_sdl2.cpp

```diff
@@ -166,10 +166,12 @@ static void gfx_sdl_init(const char *game_name, bool start_in_fullscreen, uint32
 #ifdef __SWITCH__
     // For Switch we need to set the window width before creating the window
     Ship::Switch::GetDisplaySize(&window_width, &window_height);
+    width = window_width;
+    height = window_height;
```

Comment:
@KiritoDv i think these lines should make it so the values are correct for the switch but i am not able to test it

Reply:
so would the idea be to call `Ship::Switch::GetDisplaySize` in `Window::Init` in `Window.cpp` instead?

Reply:
i like it in theory, only thing i noticed when looking through the code is that we're using `uint32_t` for `dwWidth` and `dwHeight` in `Window.cpp`, where as `window_width` and `window_height` are just `int` in `gfx_sdl2.cpp`

since we're using pointers in `Ship::Switch::GetDisplaySize`, and we still need to make the call to it in `gfx_sdl2.cpp` for `SDL_WINDOWEVENT_SIZE_CHANGED`, it seems we'd want to change something there

we also use `SDL_GL_GetDrawableSize` which uses `int` pointers (https://wiki.libsdl.org/SDL_GL_GetDrawableSize)

tl;dr - we'd need to change `dwWidth` and `dwHeight` in `Window.cpp` to be signed ints instead of `uint32_t`, if you're cool with that i'll make the change

Reply:
i'm not so sure about this now that i started on it, i pushed what needs to change for that to work here <https://github.com/briaguya-ai/Shipwright/commit/a255346d0b606d127a718129b48ab19144b41dbb>, but haven't even touched the directx side of things yet.

it feels like a lot of changes from `uint32_t` to `int` just to make `SDL_GL_GetDrawableSize` happy...